MainWindow.cpp

```diff
@@ -131,8 +131,9 @@ void MainWindow::openSubWindow(buffers::TreeNode *item) {
                                 {TypeCase::kRoom, EditorFactory<RoomEditor>},
                                 {TypeCase::kSettings, EditorFactory<SettingsEditor>}});
 
-  auto subWindow = subWindows[item];
-  if (!subWindows.contains(item) || subWindow == nullptr) {
+  auto swIt = subWindows.find(item);
+  QMdiSubWindow *subWindow = (swIt == subWindows.end()) ? nullptr : *swIt;
+  if (subWindow == nullptr) {

```

Comment:
Just check if `swIt == end()`, here, and then assign `subWindow = *swIt` below. Unless `*swIt` can be null? That would be pretty ugly. But I'd just include that in the if, here.

Reply:
You're missing the part outside the closing brace. `*swIt` can be null and this logic will still work but it won't/shouldn't be. We need `subWindow` whether it's found or not, in order to show it and set it active. When it isn't found, we attempt to open it, so in either case we have a valid `subWindow` by that point.

https://github.com/enigma-dev/RadialGM/blob/a30230a4702f34f07a8f6f605ab618b9049bd6c2/MainWindow.cpp#L154

Reply:
What I'm advising is that you check both null (`== end`) and missing (`!*it`) in this conditional, instead of doing a ternary on the line before to quietly handle part of this check. After the if, `*swIt` will be a valid pointer.

Reply:
Alright: 74979a0e67e8af343e49921ed512bf6697eabfb1

How's that? I tested it again, seems to be correct.

Reply:
Ok: 2f6aef530c127ef32c4975a63876414b056eb35b

You didn't like that, so you said to do this. Alright? We good now?

Reply:
Yes, thank you. 😛 Fire away.